### ISSUE TITLE
Trivial example does not save/restore weights, prints warnings from TensorFlow/Keras implementation

### ISSUE THREAD

**Issue by diyessi** [tensorflow/tensorflow#43777]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS 10.15.7, colab
- TensorFlow installed from (source or binary): pip, colab
- TensorFlow version (use command below): 2.3.1 (MacOS), v2.3.0-0-gb36436b087 2.3.0 (colab)
- Python version: 3.8.5 (MacOS) 3.6.9 (colab)

**Describe the current behavior**
Trained weights are not restored, two warnings about the TensorFlow implementation are printed:
`WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/training/tracking/tracking.py:111: Model.state_updates (from tensorflow.python.keras.engine.training) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.
WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/training/tracking/tracking.py:111: Layer.updates (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.`

**Describe the expected behavior**
Weights are restored (test accuracy is consistent with saved model), no warnings printed.
I find several *closed* bug reports from others over that look like the same problems  (lack of saving and the bogus warnings), going back to the 2.0.0 days. It has clearly not been fixed.

**Standalone code to reproduce the issue**

```python
import tensorflow as tf

(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
x_train = x_train.astype('float32') / 255
x_test = x_test.astype('float32') / 255

model = tf.keras.models.Sequential([
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(512, activation='relu'),
    tf.keras.layers.Dense(10, activation='softmax')])
model.compile(optimizer='rmsprop',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

model.fit(
    x_train,
    y_train,
    epochs=8,
    batch_size=128,
    validation_split=0.2)

test_loss, test_acc = model.evaluate(x_test, y_test)
print('model_test_acc:', test_acc)

model.save("my_saved_path")

saved_model = tf.keras.models.load_model("my_saved_path")

test_loss, test_acc = saved_model.evaluate(x_test, y_test)
print('saved_model_test_acc:', test_acc)
```

**Comment by ravikyram** [comment]
@diyessi 

This issue has been resolved in TF nightly version. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/db47a93deb4f8ba866bfefe2bd2ccd35/untitled411.ipynb).Please, verify once and close the issue. Thanks!

**Comment by diyessi** [comment]
Weights are properly restored, and those two warnings are gone, although two new deprecation warnings about the keras implementation have appeared that I hope you will have fixed/muted for 2.4.
